Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,123 @@ public class UserCommunicateService : IUserCommunicateService
     {
         private readonly ILogger _logger;
         private readonly IWeatherServiсe _weatherServiсe;
+        private readonly IValidator<ForecastWeatherRequestDTO> _validator;
+        private readonly Invoker _invoker;
 
-        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService)
+        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService, IValidator<ForecastWeatherRequestDTO> validator)
         {
             _logger = logger;
             _weatherServiсe = weatherService;
+            _validator = validator;
+            _invoker = new Invoker();
         }
 
-        public async Task<bool> CommunicateAsync()
+        public async Task StartUpApplicationAsync()
```

Comment:
what this method does? shows menu? launches application??? or smth else...?

Reply:
+